src/linux.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2016 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net
+// Commercial License, version 1.0 or later, or (2) The General Public License
+// (GPL), version 3, depending on which licence you accepted on initial access
+// to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project
+// generally, you agree to be bound by the terms of the MaidSafe Contributor
+// Agreement, version 1.0.
+// This, along with the Licenses can be found in the root directory of this
+// project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network
+// Software distributed under the GPL Licence is distributed on an "AS IS"
+// BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+//
+// Please review the Licences for the specific language governing permissions
+// and limitations relating to use of the SAFE Network Software.
+
+
+use std::env;
+use std::ascii::AsciiExt;
+use std::fs::{create_dir_all, File};
+use std::io::Write;
+use std::process::Command;
+use std::path::PathBuf;
+
+use app::App;
+
+
+/// Open a given URI on Linux systems
+pub fn open(uri: String) -> bool {
+    Command::new("xdg-open")
+        .arg(uri)
+        .status()
+        .expect("failed to start xdg-open")
```

Comment:
No unwraps in production - also where justified use `unwrap!()` instead as usual - but avoid in production code.
So define `system_uri::errors::SysUriError { .... }`, provide `impl Into<i32>`, start the range from `-1` and return either the error if rust code (e.g. here) or `i32` value if returning from FFI.

Check `safe_core::core::errors` for hint.

Scan the rest of the code for more of this wherever applicable